### DISCUSSION TITLE
fastq.gz has two multiplexed samples - How to best run lineage tree reconstruction on each sample?

### DISCUSSION THREAD

**Discussion by analeigh-gui** [milaboratory/mixcr#1010] [Q&A]
Hello again! I am working on BD Rhapsody full-length single-cell BCR data. 

The data comes from two samples that are multiplexed by the BD sample multiplexing kit. This means the fastq.gz files contain reads from both samples. I already processed the data and know which read is from which cell and this cell comes from which sample.

My goal is to run the lineage tree reconstruction on each sample. I am thinking about the below options:
1. Grab reads from bam based on Cell ID to reconstruct fastq.gz for each sample
2. Separate the fastq.gz based on Cell ID and follow https://mixcr.com/mixcr/reference/ref-input-file-name-expansion/

What do you think about this? Would you happen to have any recommendations on how to approach this best? Thanks!

**Comment by mizraelson** [comment]
Hi, can you please share the sample tag barcode structure? If I understand correctly you have separate fastq files that pair cell ids to samples. MiXCR can use these files to demultiplex the data.

**Reply by analeigh-gui** [reply]
Thanks for the reply! Yes, the Rhapsody has R1 having cell ids and R2 having SMK. Will sample tag barcode be enough? Or the structure of the whole read? 

I am also interested in running MIXCR pipeline on a selected number of cells. Do you have recommendation on this other than having fastq.gz files on every cell?

**Comment by PoslavskySV** [comment]
This is definitely something that can be done easily with MiXCR [sample tables](https://mixcr.com/mixcr/reference/ref-samples-table/) functionality, but we need to know the barcode structure to advice.

**Comment by analeigh-gui** [comment]
Thanks for getting back on me. Will sample tag barcode be enough? Or the structure of the whole read?

I am also interested in running MIXCR pipeline on a selected number of cells. Do you have recommendation on this other than having fastq.gz files on every cell?

**Reply by PoslavskySV** [reply]
Hi,

we need structure of the whole read to understand how to modify [tag pattern](https://mixcr.com/mixcr/reference/ref-tag-pattern/) to account for sample barcode. The current structure used for [BD](https://mixcr.com/mixcr/reference/overview-built-in-presets/#bd-rhapsody) full-length protocol is the following:

<img width="728" alt="image" src="https://user-images.githubusercontent.com/3834261/220111636-00b850be-97f7-4ab1-971d-a22c9d43b3c8.png">
